reconcile/test/test_vault_utils.py

```diff
@@ -45,3 +45,65 @@ def test_sleep_is_called(sleep):
 
         with pytest.raises(SleepCalled):
             client._auto_refresh_client_auth()
+
+
+def test_key_has_leading_space():
+    # The natural thing to do to gain access to the SecretFormatProblem custom
+    # Exception would be to rehome its import to the top of this test file.
+    # However, the TestVaultUtils class will reimport the vault library and
+    # "undo" the import. Thus, the import will be done in each test method that
+    # uses the exception.
+    from reconcile.utils.vault import SecretFormatProblem
+
+    with pytest.raises(
+        SecretFormatProblem,
+        match="Secret key has whitespace. Expected 'leading_space' but got ' leading_space'",
+    ):
+        vc = testVaultClient()
+        vc._get_mount_version_by_secret_path = MagicMock(return_value=2)
+        vc._read_all_v2 = MagicMock(
+            return_value=({" leading_space": "leadingspace"}, 2)
+        )
+
+        vc.read_all({"path": "/secret", "version": 2})
+
+
+def test_key_has_trailing_space():
+    from reconcile.utils.vault import SecretFormatProblem
+
+    with pytest.raises(
+        SecretFormatProblem,
+        match="Secret key has whitespace. Expected 'trailing_space' but got 'trailing_space '",
+    ):
+        vc = testVaultClient()
```

Comment:
Do you know https://docs.pytest.org/en/7.3.x/how-to/parametrize.html#pytest-mark-parametrize? with `pytest-mark-parametrize` you could reuse your test function.